### PR TITLE
[GITHUB] Build Release builds for all compilers

### PR DESCRIPTION
**Pull request by Extravert-ir**

### CODE REVIEW

**Comment by SergeGautherie** [comment]
What about #3825?

**Comment by learn-more** [comment]
> What about #3825?

What about it?


> What about #3825?

What about it?

**Comment by tkreuzer** [comment]
Unrelated to this change, but I noticed it's uploading ISOs. Why would we ever want to have ISOs from PRs?

**Review by tkreuzer — APPROVED** [review]

**Review by zefklop — APPROVED** [review]

**Comment by Extravert-ir** [comment]
> Unrelated to this change, but I noticed it's uploading ISOs. Why would we ever want to have ISOs from PRs?

Sometimes it may be useful for testing. But generally, I don't disable it because it costs nothing

File: sdk/cmake/msvc.cmake [hunk 1]
```diff
@@ -279,6 +283,12 @@ function(set_module_type_toolchain MODULE TYPE)
             # Mark .rsrc section as non-disposable non-pageable, as bugcheck code needs to access it
             target_link_options(${MODULE} PRIVATE /SECTION:.rsrc,!DP)
         endif()
+
+        # Clang may insert the call during some optimizations,
+        # but i386 ntoskrnl doesn't export memcmp (before NT 6.2)
+        if(CMAKE_C_COMPILER_ID STREQUAL "Clang" AND ARCH STREQUAL "i386")
+            target_compile_options(${MODULE} PRIVATE -fno-builtin-memcmp)
+        endif()
```
Comment (zefklop) [inline sdk/cmake/msvc.cmake:291]: https://git.reactos.org/?p=reactos.git;a=blob;f=ntoskrnl/CMakeLists.txt;h=6849bcf009987f137755d9a2fe26df69374d6ac0;hb=HEAD#l33

Both are fine, but we should be consistent.

File: sdk/lib/crt/msvcrtex.cmake [hunk 2]
```diff
@@ -43,19 +43,24 @@ else()
...
-    endif()
+endif()
+
+if(CMAKE_C_COMPILER_ID STREQUAL "Clang")
+    # Clang performs some optimizations requiring those funtions
+    list(APPEND MSVCRTEX_SOURCE
+        math/exp2.c
+        math/exp2f.c
+        )
 endif()
```
Comment (zefklop) [inline sdk/lib/crt/msvcrtex.cmake:54]: :+1: for CMAKE_C_COMPILER_ID

**Review by tkreuzer — APPROVED** [review]

**Comment by SergeGautherie** [comment]
> > What about #3825?
> 
> What about it?

Why ignore/hijack it?

**Pull request merged**